metagamer/environments/qagent.py

```diff
@@ -0,0 +1,144 @@
+from typing import Tuple, Optional, Any, Dict
+
+GAMMA = 0.9
+import random
+import numpy as np
+from metagamer.environments import tictactoe
+from metagamer.environments import qtable
+
+
+def exploit(epsilon):
+    """exploitation increases as epsilon decreases"""
+    return random.random() > epsilon
+
+
+class TicTacToeTableEnv(tictactoe.TicTacToeEnv):
+    """An TicTacTo subclass environment adapted to Q learning"""
+
+    SYMBOLS = {1: "X", -1: "O", 0: " "}
+
+    def _get_obs(self) -> str:
+        return "".join([self.SYMBOLS[i] for i in self.board.flatten()])
+
+    def action_to_2d(self, action: int):
```

Comment:
problem here is we have an agent that works in 1d for both but another agent that works in 2d for both. This feels like we've ended up with a bad abstraction here. 

In this case, I think the agents should handle the conversion a bit more explicitly.